userguide/tutorials/plugin_development.adoc

```diff
@@ -202,38 +202,49 @@ A bundle is nothing but a JAR file. However, its `manifest.mf` has some addition
 
-==== Importing Packages in Plugins
+==== Kill Bill OSGi Overview
 
-As explained earlier, Kill Bill plugins are packaged as OSGi bundles. The http://felix.apache.org/documentation/subprojects/apache-felix-maven-bundle-plugin-bnd.html[`maven-bundle-plugin`] specified in the https://github.com/killbill/killbill-hello-world-java-plugin/blob/3aa938d19fdfba81c7c035b45c3f17cac74db177/pom.xml[`pom.xml`] is responsible for this. Thus, the `maven-bundle-plugin` takes care of creating the jar with the correct OSGi headers (including adding the required packages to the `Import-Package` header). In addition, the https://github.com/killbill/killbill-oss-parent/blob/f3d9725cc7759a54eb6f978b3d3f763e1d170021/pom.xml[killbill-oss-parent pom] file (which is the parent of the plugin `pom.xml` file) also specifies the packages to be included in the `Import-Package` header.
+Before we dive into the details, let us understand at a high-level how the import-export mechanism works in case of the core Kill Bill system and its plugins. 
 
-Thus, when a plugin is built, the `Import-Package` header is automatically computed based on:
+* The Kill Bill core itself is packaged as an OSGi bundle (referred to as system bundle). It exports several packages. This is explained in the <<Packages exported by Kill Bill>> section. 
 
-* Packages computed by the `maven-bundle-plugin`.
-* Packages specified in the `killbill-oss-parent` pom file.
+* Packages exported by Kill Bill are automatically imported by a plugin. This is explained in the <<Packages Imported by Plugins by Default>> section. 
+
+* However, in some cases, a plugin may need to explicitly import packages exported by Kill Bill. This is explained in the <<Importing Additional Packages in Plugins>> section.
 
 
 ==== Packages exported by Kill Bill
 
-The Kill Bill system bundle also exports the packages which it desires to share with plugins. The packages exported by the Kill Bill system bundle are specified in the https://github.com/killbill/killbill-platform/blob/4e735e9c3367cee2272964f16d028795f55592d3/osgi/src/main/java/org/killbill/billing/osgi/config/OSGIConfig.java#L49[OSGIConfig.java]. Additionally, Kill Bill also offers a system property, `org.killbill.osgi.system.bundle.export.packages.extra`, to specify additional packages to be exported by the system bundle and that could in turn be imported by a plugin.
+As explained earlier, the Kill Bill system bundle exports the packages which it desires to share with plugins. The packages exported by the Kill Bill system bundle are specified in the https://github.com/killbill/killbill-platform/blob/4e735e9c3367cee2272964f16d028795f55592d3/osgi/src/main/java/org/killbill/billing/osgi/config/OSGIConfig.java#L49[OSGIConfig.java]. Additionally, Kill Bill also offers a system property, `org.killbill.osgi.system.bundle.export.packages.extra`, to specify additional packages to be exported by the system bundle and that could in turn be imported by a plugin.
```

Comment:
In most (all?) cases, `org.killbill.osgi.system.bundle.export.packages.extra` is used to export additional JVM packages (e.g. `javax.net`). This is the primary use-case for that property (the default JVM packes exported are defined here: https://github.com/killbill/killbill-platform/blob/4e735e9c3367cee2272964f16d028795f55592d3/osgi/src/main/java/org/killbill/billing/osgi/config/OSGIConfig.java#L113).